docs/source/testing.rst

```diff
@@ -0,0 +1,65 @@
+Testing Utils
+=============
+
+Base TestCases
+--------------
+
+There are two main test cases available in Sparkly:
+ - ``SparklyTest`` creates a new sparkly session per each test case.
+ - ``SparklyGlobalSessionTest`` uses a single sparkly session for all test cases to boot performance.
+
+.. code-block:: python
+
+    from sparkly import SparklySession
+    from sparkly.test import SparklyTest
+
+
+    class MyTestCase(SparklyTest):
+        session = SparklySession
+
+        def test(self):
+            df = self.spark.read_ext.by_url(...)
+            self.assertDataFrameEqual(
+                df, [('test_data', 1)], ['name', 'number']
```

Comment:
It might make it slightly clearer that the 3rd param is a list of column names if the example looked something like this:
```
self.assertDataFrameEqual(
    df, [('data_1', 1), ('data_2', 2),], ['col_name_1', 'col_name_2']
)
```